File: src/main/java/com/qtech/bubbles/common/entity/meta/MetaData.java
package com.qtech.bubbles.common.entity.meta;

import com.qtech.bubbles.common.entity.Attribute;
import com.qtech.bubbles.common.entity.Entity;
import com.qtech.bubbles.common.entity.Modifier;
import org.jetbrains.annotations.Nullable;

import java.util.HashMap;

@Deprecated
@SuppressWarnings("unchecked")
public class MetaData {
    private final Entity entity;
    private boolean invulnerable;

    private final HashMap<Modifier, Double> modifiers = new HashMap<>();
    private final HashMap<Attribute, Object> attributes = new HashMap<>();

    public MetaData(Entity entity) {
        this.entity = entity;
    }

    public Entity getEntity() {
        return entity;
    }

    public boolean isInvulnerable() {
        return invulnerable;
    }

    public void setInvulnerable(boolean invulnerable) {
        this.invulnerable = invulnerable;
    }

    public void setModifier(Modifier modifier, double value) {
        modifiers.put(modifier, value);
    }

    public double getModifier(Modifier modifier) {
        double out = 0.0d;
        Double val = modifiers.get(modifier);
        if (val != null) {
            out = val;
        }
        return out;
    }

    public <T> void setAttribute(Attribute attribute, T value) {
        attributes.put(attribute, value);
    }

    @Nullable
    public <T> T getAttribute(Attribute attribute) {
        return (T) attributes.get(attribute);
    }
}

File: src/main/kotlin/qtech/bubbles/common/entity/meta/MetaData.kt
package qtech.bubbles.common.entity.meta

import qtech.bubbles.common.entity.Attribute
import qtech.bubbles.entity.Entity
import qtech.bubbles.common.entity.Modifier

@Deprecated("")
class MetaData(val entity: Entity) {
    var isInvulnerable = false
    private val modifiers = HashMap<Modifier, Double>()
    private val attributes = HashMap<Attribute, Any>()
    fun setModifier(modifier: Modifier, value: Double) {
        modifiers[modifier] = value
    }

    fun getModifier(modifier: Modifier): Double {
        var out = 0.0
        val `val` = modifiers[modifier]
        if (`val` != null) {
            out = `val`
        }
        return out
    }

    fun <T> setAttribute(attribute: Attribute, value: T) {
        attributes[attribute] = value!!
    }

    @Suppress("UNCHECKED_CAST")
    fun <T> getAttribute(attribute: Attribute): T? {
        return attributes[attribute] as T?
    }
}